controllers/topology/internal/mergepatch/mergepatch.go

```diff
@@ -193,6 +237,43 @@ func removePath(patchMap map[string]interface{}, path contract.Path) {
 	}
 }
 
+// enforcePath a path from authoritativeMap into the twoWayMap thus
+// enforcing changes aligned to the modified object for the authoritative paths.
+func enforcePath(authoritativeMap, twoWayMap map[string]interface{}, path contract.Path) {
```

Comment:
```suggestion
func enforcePath(authoritative, twoWay map[string]interface{}, path contract.Path) {
```

Reply:
If I understood this function correctly, we always want to make sure that a certain path from the authority is always set in the twoWayMap, couldn't we use something like: https://github.com/kubernetes/kubernetes/blob/bafa87c553b1ad2700eaa0b3808aa7c6b6ce1a90/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/helpers.go#L208

Reply:
Semantically yes, but with the current implementation  this is not practical because the patch is a map we we should coordinate enforce paths with allow paths and ignore paths.
Hopefully all this will be greatly simplified soon by implementing https://github.com/kubernetes-sigs/cluster-api/issues/5598